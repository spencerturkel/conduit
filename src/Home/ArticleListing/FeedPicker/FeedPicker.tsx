import * as React from 'react';
import {StatelessComponent} from 'react';

interface LoggedInFeedPickerProps {
    activeFeed: 'user' | 'global';
    onGlobalFeedClicked: () => void;
    onUserFeedClicked: () => void;
    loggedIn: true;
}

interface LoggedOutFeedPickerProps {
    activeFeed: 'global';
    loggedIn: false;
}

export type FeedPickerProps = LoggedInFeedPickerProps | LoggedOutFeedPickerProps;

const FeedPicker = ((props: FeedPickerProps) => (
    <div className="feed-toggle">
        <ul className="nav nav-pills outline-active">
            <li className="nav-item">
                <a
                    className={`nav-link ${!props.loggedIn ? 'disabled' : props.activeFeed === 'user' ? 'active' : ''}`}
                    onClick={e => {
                        e.preventDefault();

                        if (props.loggedIn) {
                            props.onUserFeedClicked();
                        }
                    }}>
                    Your Feed
                </a>
            </li>
            <li className="nav-item">
                <a
                    className={`nav-link ${props.activeFeed === 'global' ? 'active' : ''}`}
                    onClick={e => {
                        e.preventDefault();

                        if (props.loggedIn) {
                            props.onGlobalFeedClicked();
                        }
                    }}>
                    Global Feed
                </a>
            </li>
        </ul>
    </div>
)) as StatelessComponent<FeedPickerProps>;

export default FeedPicker;
