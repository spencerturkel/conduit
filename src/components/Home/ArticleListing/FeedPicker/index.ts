export { default } from "./FeedPicker";
export * from "./FeedPicker";
