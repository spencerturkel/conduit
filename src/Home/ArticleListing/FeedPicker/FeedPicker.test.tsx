import {mount, ReactWrapper} from 'enzyme';
import * as React from 'react';

import FeedPicker from './FeedPicker';

let sut: ReactWrapper;

beforeEach(() => {
    sut = mount(<FeedPicker />);
});

it('should match snapshot', () => {
    expect(sut).toMatchSnapshot();
});
