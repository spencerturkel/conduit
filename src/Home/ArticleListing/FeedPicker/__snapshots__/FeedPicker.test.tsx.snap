// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <FeedPicker
    activeFeed="global"
    loggedIn={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeFeed": "global",
      "loggedIn": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="feed-toggle"
>
        <ul
                class="nav nav-pills outline-active"
        >
                <li
                        class="nav-item"
                >
                        <a
                                class="nav-link disabled"
                        >
                                Your Feed
                        </a>
                </li>
                <li
                        class="nav-item"
                >
                        <a
                                class="nav-link active"
                        >
                                Global Feed
                        </a>
                </li>
        </ul>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav nav-pills outline-active"
>
          <li
                    className="nav-item"
          >
                    <a
                              className="nav-link disabled"
                              onClick={[Function]}
                    >
                              Your Feed
                    </a>
          </li>
          <li
                    className="nav-item"
          >
                    <a
                              className="nav-link active"
                              onClick={[Function]}
                    >
                              Global Feed
                    </a>
          </li>
</ul>,
        "className": "feed-toggle",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <ul
            class="nav nav-pills outline-active"
>
            <li
                        class="nav-item"
            >
                        <a
                                    class="nav-link disabled"
                        >
                                    Your Feed
                        </a>
            </li>
            <li
                        class="nav-item"
            >
                        <a
                                    class="nav-link active"
                        >
                                    Global Feed
                        </a>
            </li>
</ul>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-item"
>
                <a
                                className="nav-link disabled"
                                onClick={[Function]}
                >
                                Your Feed
                </a>
</li>,
              <li
                className="nav-item"
>
                <a
                                className="nav-link active"
                                onClick={[Function]}
                >
                                Global Feed
                </a>
</li>,
            ],
            "className": "nav nav-pills outline-active",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <li
                class="nav-item"
>
                <a
                                class="nav-link disabled"
                >
                                Your Feed
                </a>
</li>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link disabled"
                  onClick={[Function]}
>
                  Your Feed
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    class="nav-link disabled"
>
                    Your Feed
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Your Feed",
                    "className": "nav-link disabled",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Your Feed",
                  ],
                  "type": "a",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": <li
                class="nav-item"
>
                <a
                                class="nav-link active"
                >
                                Global Feed
                </a>
</li>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link active"
                  onClick={[Function]}
>
                  Global Feed
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    class="nav-link active"
>
                    Global Feed
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Global Feed",
                    "className": "nav-link active",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Global Feed",
                  ],
                  "type": "a",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeFeed": "global",
        "loggedIn": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="feed-toggle"
>
          <ul
                    class="nav nav-pills outline-active"
          >
                    <li
                              class="nav-item"
                    >
                              <a
                                        class="nav-link disabled"
                              >
                                        Your Feed
                              </a>
                    </li>
                    <li
                              class="nav-item"
                    >
                              <a
                                        class="nav-link active"
                              >
                                        Global Feed
                              </a>
                    </li>
          </ul>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="nav nav-pills outline-active"
>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link disabled"
                                    onClick={[Function]}
                        >
                                    Your Feed
                        </a>
            </li>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link active"
                                    onClick={[Function]}
                        >
                                    Global Feed
                        </a>
            </li>
</ul>,
          "className": "feed-toggle",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <ul
              class="nav nav-pills outline-active"
>
              <li
                            class="nav-item"
              >
                            <a
                                          class="nav-link disabled"
                            >
                                          Your Feed
                            </a>
              </li>
              <li
                            class="nav-item"
              >
                            <a
                                          class="nav-link active"
                            >
                                          Global Feed
                            </a>
              </li>
</ul>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link disabled"
                                    onClick={[Function]}
                  >
                                    Your Feed
                  </a>
</li>,
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link active"
                                    onClick={[Function]}
                  >
                                    Global Feed
                  </a>
</li>,
              ],
              "className": "nav nav-pills outline-active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <li
                  class="nav-item"
>
                  <a
                                    class="nav-link disabled"
                  >
                                    Your Feed
                  </a>
</li>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link disabled"
                    onClick={[Function]}
>
                    Your Feed
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      class="nav-link disabled"
>
                      Your Feed
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Your Feed",
                      "className": "nav-link disabled",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Your Feed",
                    ],
                    "type": "a",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": <li
                  class="nav-item"
>
                  <a
                                    class="nav-link active"
                  >
                                    Global Feed
                  </a>
</li>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link active"
                    onClick={[Function]}
>
                    Global Feed
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      class="nav-link active"
>
                      Global Feed
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Global Feed",
                      "className": "nav-link active",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Global Feed",
                    ],
                    "type": "a",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should match snapshots 2`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <FeedPicker
    activeFeed="global"
    loggedIn={true}
    onGlobalFeedClicked={[Function]}
    onUserFeedClicked={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeFeed": "global",
      "loggedIn": true,
      "onGlobalFeedClicked": [Function],
      "onUserFeedClicked": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="feed-toggle"
>
        <ul
                class="nav nav-pills outline-active"
        >
                <li
                        class="nav-item"
                >
                        <a
                                class="nav-link "
                        >
                                Your Feed
                        </a>
                </li>
                <li
                        class="nav-item"
                >
                        <a
                                class="nav-link active"
                        >
                                Global Feed
                        </a>
                </li>
        </ul>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav nav-pills outline-active"
>
          <li
                    className="nav-item"
          >
                    <a
                              className="nav-link "
                              onClick={[Function]}
                    >
                              Your Feed
                    </a>
          </li>
          <li
                    className="nav-item"
          >
                    <a
                              className="nav-link active"
                              onClick={[Function]}
                    >
                              Global Feed
                    </a>
          </li>
</ul>,
        "className": "feed-toggle",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <ul
            class="nav nav-pills outline-active"
>
            <li
                        class="nav-item"
            >
                        <a
                                    class="nav-link "
                        >
                                    Your Feed
                        </a>
            </li>
            <li
                        class="nav-item"
            >
                        <a
                                    class="nav-link active"
                        >
                                    Global Feed
                        </a>
            </li>
</ul>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-item"
>
                <a
                                className="nav-link "
                                onClick={[Function]}
                >
                                Your Feed
                </a>
</li>,
              <li
                className="nav-item"
>
                <a
                                className="nav-link active"
                                onClick={[Function]}
                >
                                Global Feed
                </a>
</li>,
            ],
            "className": "nav nav-pills outline-active",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <li
                class="nav-item"
>
                <a
                                class="nav-link "
                >
                                Your Feed
                </a>
</li>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link "
                  onClick={[Function]}
>
                  Your Feed
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    class="nav-link "
>
                    Your Feed
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Your Feed",
                    "className": "nav-link ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Your Feed",
                  ],
                  "type": "a",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": <li
                class="nav-item"
>
                <a
                                class="nav-link active"
                >
                                Global Feed
                </a>
</li>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link active"
                  onClick={[Function]}
>
                  Global Feed
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    class="nav-link active"
>
                    Global Feed
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Global Feed",
                    "className": "nav-link active",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Global Feed",
                  ],
                  "type": "a",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeFeed": "global",
        "loggedIn": true,
        "onGlobalFeedClicked": [Function],
        "onUserFeedClicked": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="feed-toggle"
>
          <ul
                    class="nav nav-pills outline-active"
          >
                    <li
                              class="nav-item"
                    >
                              <a
                                        class="nav-link "
                              >
                                        Your Feed
                              </a>
                    </li>
                    <li
                              class="nav-item"
                    >
                              <a
                                        class="nav-link active"
                              >
                                        Global Feed
                              </a>
                    </li>
          </ul>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="nav nav-pills outline-active"
>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link "
                                    onClick={[Function]}
                        >
                                    Your Feed
                        </a>
            </li>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link active"
                                    onClick={[Function]}
                        >
                                    Global Feed
                        </a>
            </li>
</ul>,
          "className": "feed-toggle",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <ul
              class="nav nav-pills outline-active"
>
              <li
                            class="nav-item"
              >
                            <a
                                          class="nav-link "
                            >
                                          Your Feed
                            </a>
              </li>
              <li
                            class="nav-item"
              >
                            <a
                                          class="nav-link active"
                            >
                                          Global Feed
                            </a>
              </li>
</ul>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link "
                                    onClick={[Function]}
                  >
                                    Your Feed
                  </a>
</li>,
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link active"
                                    onClick={[Function]}
                  >
                                    Global Feed
                  </a>
</li>,
              ],
              "className": "nav nav-pills outline-active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <li
                  class="nav-item"
>
                  <a
                                    class="nav-link "
                  >
                                    Your Feed
                  </a>
</li>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link "
                    onClick={[Function]}
>
                    Your Feed
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      class="nav-link "
>
                      Your Feed
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Your Feed",
                      "className": "nav-link ",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Your Feed",
                    ],
                    "type": "a",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": <li
                  class="nav-item"
>
                  <a
                                    class="nav-link active"
                  >
                                    Global Feed
                  </a>
</li>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link active"
                    onClick={[Function]}
>
                    Global Feed
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      class="nav-link active"
>
                      Global Feed
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Global Feed",
                      "className": "nav-link active",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Global Feed",
                    ],
                    "type": "a",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should match snapshots 3`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <FeedPicker
    activeFeed="user"
    loggedIn={true}
    onGlobalFeedClicked={[Function]}
    onUserFeedClicked={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeFeed": "user",
      "loggedIn": true,
      "onGlobalFeedClicked": [Function],
      "onUserFeedClicked": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="feed-toggle"
>
        <ul
                class="nav nav-pills outline-active"
        >
                <li
                        class="nav-item"
                >
                        <a
                                class="nav-link active"
                        >
                                Your Feed
                        </a>
                </li>
                <li
                        class="nav-item"
                >
                        <a
                                class="nav-link "
                        >
                                Global Feed
                        </a>
                </li>
        </ul>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav nav-pills outline-active"
>
          <li
                    className="nav-item"
          >
                    <a
                              className="nav-link active"
                              onClick={[Function]}
                    >
                              Your Feed
                    </a>
          </li>
          <li
                    className="nav-item"
          >
                    <a
                              className="nav-link "
                              onClick={[Function]}
                    >
                              Global Feed
                    </a>
          </li>
</ul>,
        "className": "feed-toggle",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <ul
            class="nav nav-pills outline-active"
>
            <li
                        class="nav-item"
            >
                        <a
                                    class="nav-link active"
                        >
                                    Your Feed
                        </a>
            </li>
            <li
                        class="nav-item"
            >
                        <a
                                    class="nav-link "
                        >
                                    Global Feed
                        </a>
            </li>
</ul>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-item"
>
                <a
                                className="nav-link active"
                                onClick={[Function]}
                >
                                Your Feed
                </a>
</li>,
              <li
                className="nav-item"
>
                <a
                                className="nav-link "
                                onClick={[Function]}
                >
                                Global Feed
                </a>
</li>,
            ],
            "className": "nav nav-pills outline-active",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <li
                class="nav-item"
>
                <a
                                class="nav-link active"
                >
                                Your Feed
                </a>
</li>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link active"
                  onClick={[Function]}
>
                  Your Feed
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    class="nav-link active"
>
                    Your Feed
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Your Feed",
                    "className": "nav-link active",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Your Feed",
                  ],
                  "type": "a",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": <li
                class="nav-item"
>
                <a
                                class="nav-link "
                >
                                Global Feed
                </a>
</li>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="nav-link "
                  onClick={[Function]}
>
                  Global Feed
</a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    class="nav-link "
>
                    Global Feed
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Global Feed",
                    "className": "nav-link ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Global Feed",
                  ],
                  "type": "a",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeFeed": "user",
        "loggedIn": true,
        "onGlobalFeedClicked": [Function],
        "onUserFeedClicked": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="feed-toggle"
>
          <ul
                    class="nav nav-pills outline-active"
          >
                    <li
                              class="nav-item"
                    >
                              <a
                                        class="nav-link active"
                              >
                                        Your Feed
                              </a>
                    </li>
                    <li
                              class="nav-item"
                    >
                              <a
                                        class="nav-link "
                              >
                                        Global Feed
                              </a>
                    </li>
          </ul>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="nav nav-pills outline-active"
>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link active"
                                    onClick={[Function]}
                        >
                                    Your Feed
                        </a>
            </li>
            <li
                        className="nav-item"
            >
                        <a
                                    className="nav-link "
                                    onClick={[Function]}
                        >
                                    Global Feed
                        </a>
            </li>
</ul>,
          "className": "feed-toggle",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <ul
              class="nav nav-pills outline-active"
>
              <li
                            class="nav-item"
              >
                            <a
                                          class="nav-link active"
                            >
                                          Your Feed
                            </a>
              </li>
              <li
                            class="nav-item"
              >
                            <a
                                          class="nav-link "
                            >
                                          Global Feed
                            </a>
              </li>
</ul>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link active"
                                    onClick={[Function]}
                  >
                                    Your Feed
                  </a>
</li>,
                <li
                  className="nav-item"
>
                  <a
                                    className="nav-link "
                                    onClick={[Function]}
                  >
                                    Global Feed
                  </a>
</li>,
              ],
              "className": "nav nav-pills outline-active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <li
                  class="nav-item"
>
                  <a
                                    class="nav-link active"
                  >
                                    Your Feed
                  </a>
</li>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link active"
                    onClick={[Function]}
>
                    Your Feed
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      class="nav-link active"
>
                      Your Feed
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Your Feed",
                      "className": "nav-link active",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Your Feed",
                    ],
                    "type": "a",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": <li
                  class="nav-item"
>
                  <a
                                    class="nav-link "
                  >
                                    Global Feed
                  </a>
</li>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="nav-link "
                    onClick={[Function]}
>
                    Global Feed
</a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      class="nav-link "
>
                      Global Feed
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Global Feed",
                      "className": "nav-link ",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Global Feed",
                    ],
                    "type": "a",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
